### PR TITLE
Implement building location detection

### LINKED ISSUE #16: US-3: Building location
As a student, I want to know which building I am currently located in so that I know where I am situated.

**ACCEPTANCE CRITERIA:**
* Have a crosshair button that, when pressed, shows the building that I'm currently in
* Must match [Mockup](https://www.figma.com/file/1XeyBVJ2XyC6AHDHYwxvdG/WayFinder-SOEN390?node-id=46%3A2)
* Write unit tests and component tests

**USER FLOW ACCEPTANCE TEST:**
1. Press on the crosshair button
2. Map centers to the user location
3. User is informed if they are in a campus building or not
4. If user is in a building, then building info is shown

### PR DESCRIPTION
**Pull request by cristian-aldea**
A new component, the building location button, was implemented, allowing
the user to locate themselves on the campus.

The liberty was taken to add type information for relevant methods and
fields in the affected code

Also, the liberty was taken to update all dependencies to their latest
compatible versions.

### CODE REVIEW

File: src/screens/map-screen/map-screen.component.tsx [hunk 1]
```diff
@@ -42,12 +51,57 @@ const MapScreen = () => {
...
 
+  /*
+   * Handles the event when the user pressed on the Building Location Button
+   *
+   * Sets the current location to whereever the user currently is, and then
+   * perform point-polygon collision detection to find which building the
+   * user is in.
+   */
+  const onBuildingLocationPress = (): void => {
+    getCurrentLocationAsync().then(response => {
```
Comment (cristian-aldea) [inline src/screens/map-screen/map-screen.component.tsx:68]: to temporarily test what would have happened if you were in a building, simply do the following:
```
getCurrentLocationAsync().then(response => {
      response.coords.latitude = [lat];
      response.coords.longitude = [long];
      // Set current location
      setCurrentLocation({
        latitude: response.coords.latitude,
        longitude: response.coords.longitude
      });
...
```
where [lat] and [long] are coordinate within a given building

File: src/screens/map-screen/map-screen.component.tsx [hunk 2]
```diff
@@ -42,12 +51,57 @@ const MapScreen = () => {
...
+      // Attemp to find the building the user is in.
+      let inBuilding = false;
+      Object.keys(BuildingCoordinates).forEach(building => {
+        const coordinates = BuildingCoordinates[building];
+        if (isPointInPolygon(response.coords, coordinates)) {
+          showMessage({
+            message: `You're currently in the ${building} building!`,
+            type: "info"
+          });
+          onDisplayBuilding(true, building);
```
Comment (cristian-aldea) [inline src/screens/map-screen/map-screen.component.tsx:91]: call to onDisplayBuilding(true, building) might need to be changed when #19 rebased on top of these changes

**Review by RGPosadas — APPROVED** [review]
### DoD
- [x] Documentation for every component and function in your code
- [x] Implement unit and snapshot tests for every new component/function you create
- [x] Implement unit and snapshot tests for every component/function you changed which didn't have existing tests
- [x] Run all tests and make sure they pass
- [x] Final implementation should match the corresponding mockup

Note: I am aware until that #37 is merged, we cannot have unit tests. However, this should not be a blocker since the implementation is good and the snapshot test passes.

### AC
- [x] Have a crosshair button that, when pressed, shows the building that I'm currently in

### USER FLOW ACCEPTANCE TEST:
1. Press on the crosshair button
2. Map centers to the user location
3. User is informed if they are in a campus building or not
4. If user is in a building, then building info is shown
- [x] USER FLOW STATUS: PASS

**Review by bit172 — APPROVED** [review]
### DoD
- [x] Documentation for every component and function in your code
- [x] Implement unit and snapshot tests for every new component/function you create
- [x] Implement unit and snapshot tests for every component/function you changed which didn't have existing tests
- [x] Run all tests and make sure they pass
- [x] Final implementation should match the corresponding mockup

### AC
- [x] Have a crosshair button that, when pressed, shows the building that I'm currently in

### USER FLOW ACCEPTANCE TEST:
1. Press on the crosshair button
2. Map centers to the user location
3. User is informed if they are in a campus building or not
4. If user is in a building, then building info is shown
- [x] USER FLOW STATUS: PASS

LGTM

**Pull request merged**